src/server/plugins/engine/models/SummaryViewModel.ts

```diff
@@ -190,6 +166,51 @@ export class SummaryViewModel {
   }
 }
 
+function addRepeaterItem(
+  page: RepeatPageController,
+  request: Request,
+  model: FormModel,
+  state: FormSubmissionState,
+  items: DetailItem[]
+) {
+  const { options } = page.repeat
+  const { name, title } = options
+  const repeatSummaryPath = page.getSummaryPath(request)
+  const path = `/${model.basePath}${page.path}`
+  const rawValue = state[options.name]
+  const isInitialised = Array.isArray(rawValue)
```

Comment:
Wondering if it could ever be an empty array?

If we could make your class method public it's handled in `getListFromState()`

```suggestion
  const rawValue = page.getListFromState(state)
  const isInitialised = !!rawValue.length
```

Reply:
If someone were to land at the repeater summary page directly without adding any items then yes, it could be empty. If they arrived there as a new user/uninitialised session then it could also be undefined (isInitialised would be false)

Reply:
Thanks for clarifying. There's no `null` ever?

Would you mind if we used `page.getListFromState(state)` to give us a single "get a list" approach?

Reply:
I think that'd be good. I'm personally not a fan of accessing the raw state directly where it can be avoided, moving the access into a single function where the logic can be reused would be handy.

Not a blocker though, we do this all over the place as it is.

Reply:
Plus it helps me out for Monday, where I've already set up a single method for `rawValue` 😆

Reply:
> Thanks for clarifying. There's no `null` ever?

No definitely no `null`. We've been asked not to support optional repeaters (`options.min` should be positive), I'd like that to change and if it does, we'll probably choose `[]` not `null` as a marker in state to flag that the repeater page has been visited, but 0 items were added.

Reply:
Resolved with a6e0ae3b2e7ef65b7560bb0b5a39d0a9b49dc016